### ISSUE TITLE
404 not found

### ISSUE THREAD

**Issue by SeVeNDuS** [i18next/i18next-express-middleware#55]
Hi,
I am trying to load translation files from browser served by node.js but after follow all steps I always get 404 error on path like /locales/es/translation.json

Could you help me?

Regards

**Comment by jamuhl** [comment]
do you serve the locales folder? eg. via express statics?

or use: https://github.com/i18next/i18next-express-middleware#add-routes resourceHandler and change client to that load path and enable multiload

**Comment by SeVeNDuS** [comment]
I suppose that not needed to put on an static folder such as older verions. I have done and now I can load it individually but with multiload I get the following cliente error:

i18next.min.js:1 i18next::backendConnector: loading namespace translation for language es via multiloading failed

Regards

**Comment by jamuhl** [comment]
you will need to set loadPath in xhr backend to route speciefied in middleware

**Comment by SeVeNDuS** [comment]
I have the default "path" /locales/resources.json, it answer but this "{}" so after that fires the error that I told you before.

Thx!

**Comment by jamuhl** [comment]
did you add  https://github.com/i18next/i18next-express-middleware#add-routes resourceHandler 

else please paste your client config and server.js where you config i18next middleware

**Comment by SeVeNDuS** [comment]
Browser
`
var i18nextBackendOptions = {
                    allowMultiLoading: true
                };
                var i18nextOptions = {
                    lngs: ['es', 'de', 'en', 'pt', 'it', 'fr', 'es-ca'],
                    whitelist: ['es', 'de', 'en', 'pt', 'it', 'fr', 'es-ca'],
                    fallbackLng: 'en',
                    lowerCaseLng: true,
                    backend: i18nextBackendOptions
                };
                window.i18next
                    .use(window.i18nextXHRBackend)
                    .use(window.i18nextBrowserLanguageDetector)
                    .use(window.i18nextLocalStorageCache)
                    .init(i18nextOptions, function () {
                        jqueryI18next.init(window.i18next, $);
                        
//CODE
                    }
                );
`

SERVER
`
var i18nextBackendOptions = {  
	  	loadPath: __dirname + '/public/locales/{{lng}}/{{ns}}.json',
	  	addPath: __dirname + '/public/locales/{{lng}}/{{ns}}.missing.json',
	  	jsonIndent: 4
	};

	var i18nextOptions = {
		lngs: ['es', 'de', 'en', 'pt', 'it', 'fr', 'es-ca'],
		whitelist: ['es', 'de', 'en', 'pt', 'it', 'fr', 'es-ca'],
		fallbackLng: 'en',
		backend: i18nextBackendOptions,
		lowerCaseLng: true
	};

	i18next.use(i18nextBackend)
		.use(i18nextMiddleware.LanguageDetector)
		.init(i18nextOptions);

	var app = express();
	app.use(express.static('public'));

	app.post('/locales/add/:lng/:ns', i18nextMiddleware.missingKeyHandler(i18next));
	app.get('/locales/resources.json', i18nextMiddleware.getResourcesHandler(i18next));

	app.use(i18nextMiddleware.handle(i18next, {
		removeLngFromUrl: false
	}));
`

**Comment by jamuhl** [comment]
add to i18nextBackendOptions: https://github.com/i18next/i18next-xhr-backend#backend-options

```
var i18nextBackendOptions = {
  allowMultiLoading: true,
  loadPath: '/locales/resources.json?lng={{lng}}&ns={{ns}}',
};
```

what do you want to archive with lngs in init options?

```
var i18nextOptions = {
        lngs: ['es', 'de', 'en', 'pt', 'it', 'fr', 'es-ca'],
```

there is no lngs options -> lng will set lng, preload will preload and whitellist will whitelist

**Comment by Ancinek** [comment]
Hello there. I've encountered pretty similar problem. In my case I get `POST http://localhost:3005/locales/add/en/translations 404 (Not Found)`.

My server code looks like this:

```
i18n
  .use(Backend)
  .use(i18nextMiddleware.LanguageDetector)
  .init({
    preload: ['en', 'de'],
    backend: {
      loadPath: pathResolve(appRootDir.get(), 'locales/{{lng}}/{{ns}}.json'),
      addPath: pathResolve(appRootDir.get(), 'locales/{{lng}}/{{ns}}.missing.json'),
    },
    fallbackLng: 'en',
    saveMissing: true,
  }, () => {
    app
      .disable('x-powered-by')
      .use(i18nextMiddleware.handle(i18n))
      .use(cookiesMiddleware())
      .get('*', reactApplication)
      .use('/locales', express.static(pathResolve(appRootDir.get(), 'locales')))
      .use(...errorHandlers);
  });

// Listener
const port = config('port');
const listener = express()
  .use((req, res) => app.handle(req, res))
  .listen(port, () => {
    console.log(`Server listening on port ${port}.`);
  });
```

What am I missing/ doing wrong here? I got this concept from [here](https://github.com/i18next/react-i18next/blob/master/example/razzle-ssr/src/server.js)

**Comment by jamuhl** [comment]
@Ancinek you get that error in client? or on server?

looks like in client and that the missing path is not bound: https://github.com/i18next/i18next-express-middleware#add-routes

**Comment by Ancinek** [comment]
I edited it so it looks like this now:

```
i18n
  .use(Backend)
  .use(i18nextMiddleware.LanguageDetector)
  .init({
    preload: ['en', 'de'],
    backend: {
      loadPath: pathResolve(appRootDir.get(), 'locales/{{lng}}/{{ns}}.json'),
      addPath: pathResolve(appRootDir.get(), 'locales/{{lng}}/{{ns}}.missing.json'),
      allowMultiloading: true,
    },
    fallbackLng: 'en',
  }, () => {
    app
      .disable('x-powered-by')
      .use(i18nextMiddleware.handle(i18n))
      .use(cookiesMiddleware())
      .use('/locales', express.static(pathResolve(appRootDir.get(), 'locales')))
      .get('*', reactApplication)
      .post('/locales/add/:lng/:ns', i18nextMiddleware.missingKeyHandler(i18n))
      .get('/locales/:lng/:ns', i18nextMiddleware.getResourcesHandler(i18n))
      .use(...errorHandlers);
  });
```

It seemed to work when I'm in 'one level deep' path (like `localhost:3000/firstLevel`), but when on for example `localhost:3000/firstLevel/secondLevel` it tries to POST to `localhost:3000/firstLevel/locales/add/:lng/:ns` instead of omitting that firstLevel...

**Comment by jamuhl** [comment]
seems we got https://github.com/i18next/i18next-xhr-backend/blob/master/src/index.js#L7 that path wrong -> just add that addPath with starting `/` for now on init of client backend -> will do a fix for the backend asap

**Comment by Ancinek** [comment]
Great! Thanks for the fast response! :)

**Comment by jamuhl** [comment]
should be fixed in i18next-xhr-backend@1.5.1
